src/main/java/org/prebid/server/bidder/outbrain/OutbrainBidder.java

```diff
@@ -137,31 +145,89 @@ private static App updateApp(App app, Publisher publisher) {
 
     @Override
     public final Result<List<BidderBid>> makeBids(HttpCall<BidRequest> httpCall, BidRequest bidRequest) {
+        final List<BidderError> errors = new ArrayList<>();
+
         try {
             final BidResponse bidResponse = mapper.decodeValue(httpCall.getResponse().getBody(), BidResponse.class);
-            return Result.of(extractBids(httpCall.getRequest().getPayload(), bidResponse), Collections.emptyList());
+            return Result.of(extractBids(httpCall.getRequest().getPayload(), bidResponse, errors), errors);
         } catch (DecodeException | PreBidException e) {
             return Result.withError(BidderError.badServerResponse(e.getMessage()));
         }
     }
 
-    private static List<BidderBid> extractBids(BidRequest bidRequest, BidResponse bidResponse) {
+    private List<BidderBid> extractBids(BidRequest bidRequest, BidResponse bidResponse, List<BidderError> errors) {
         if (bidResponse == null || CollectionUtils.isEmpty(bidResponse.getSeatbid())) {
             return Collections.emptyList();
         }
-        return bidsFromResponse(bidRequest, bidResponse);
+        return bidsFromResponse(bidRequest, bidResponse, errors);
     }
 
-    private static List<BidderBid> bidsFromResponse(BidRequest bidRequest, BidResponse bidResponse) {
+    private List<BidderBid> bidsFromResponse(BidRequest bidRequest, BidResponse bidResponse, List<BidderError> errors) {
         return bidResponse.getSeatbid().stream()
                 .filter(Objects::nonNull)
                 .map(SeatBid::getBid)
                 .filter(Objects::nonNull)
                 .flatMap(Collection::stream)
-                .map(bid -> BidderBid.of(bid, getBidType(bid.getImpid(), bidRequest.getImp()), bidResponse.getCur()))
+                .map(bid -> createBidderBid(bid, getBidType(bid.getImpid(),
+                        bidRequest.getImp()), bidResponse.getCur(), errors))
                 .collect(Collectors.toList());
     }
 
+    private BidderBid createBidderBid(Bid bid, BidType bidType, String cur, List<BidderError> errors) {
+        if (!bidType.equals(BidType.xNative) || StringUtils.isEmpty(bid.getAdm())) {
+            return BidderBid.of(bid, bidType, cur);
+        }
+        final String resolvedAdm = resolveBidAdm(bid.getAdm(), errors);
+
+        if (resolvedAdm != null) {
+            return BidderBid.of(bid.toBuilder().adm(resolvedAdm).build(), bidType, cur);
+        }
+
+        return BidderBid.of(bid, bidType, cur);
+    }
```

Comment:
Let's refactor this section.
1. Logically it would be better if bidType will be resolved in createBidderBid, but in bidder bid we have many duplicate code, so my proposal to do something like this
```java
.map(bid -> createBidderBid(bid, bidRequest.getImp(), bidResponse.getCur(), errors))
.collect(Collectors.toList());
    }

private BidderBid createBidderBid(Bid bid, List<Imp> requestImps, String cur, List<BidderError> errors) {
        final BidType bidType = getBidType(bid.getImpid(), requestImps);

        return BidderBid.of(updateBid(bid, bidType, errors), bidType, cur);
    }

private Bid updateBid(Bid bid, BidType bidType, List<BidderError> errors) {
        final String bidAdm = bid.getAdm();
        final String resolvedAdm = bidType.equals(BidType.xNative) && StringUtils.isNotEmpty(bidAdm)
                ? resolveBidAdm(bidAdm, errors)
                : null;

        return resolvedAdm != null ? bid.toBuilder().adm(resolvedAdm).build() : bid;
    }
```